### PR TITLE
feat: add type restrictions to json syntax

### PR DESCRIPTION
**Pull request by rhamzeh**
## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: stores/gdrive/authorization-model.json [hunk 1]
```diff
@@ -148,6 +185,34 @@
...
+                "type": "folder"
+            }]
+          },
+          "can_read": { "directly_related_user_types": [] },
+          "can_share": { "directly_related_user_types": [] },
+          "viewer": {
+            "directly_related_user_types": [{
+                "type": "user"
+            }, {
+                "type": "user:*"
```
Comment (adriantam) [inline stores/gdrive/authorization-model.json:208]: I supposed we don't need user:* now, right?

File: stores/github/authorization-model.json [hunk 2]
```diff
@@ -151,6 +216,69 @@
         "repo_writer": {
           "this": {}
         }
+      },
+      "metadata": {
+        "relations": {
+          "member": {
+            "directly_related_user_types": [
+              {
+                "type": "organization"
```
Comment (adriantam) [inline stores/github/authorization-model.json:225]: I am confused - shouldn't organization member be user?

File: stores/github/authorization-model.json [hunk 3]
```diff
@@ -151,6 +216,69 @@
...
+              }
+            ]
+          },
+          "owner": {
+            "directly_related_user_types": [
+              {
+                "type": "user"
```
Comment (adriantam) [inline stores/github/authorization-model.json:232]: should owner be `organization`?

File: stores/github/authorization-model.json [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
         "member": {
           "this": {}
         }
+      },
+      "metadata": {
+        "relations": {
+          "member": {
+            "directly_related_user_types": [
+              {
+                "type": "user"
```
Comment (adriantam) [inline stores/github/authorization-model.json:19]: should team:member be directly related to organization:member?

File: stores/github/authorization-model.json (hunk 3 above)
Comment (jon-whit) [reply stores/github/authorization-model.json:232]: I believe one or more users own an org. I don't think orgs own an org, but I may be wrong.

Comment (rhamzeh) [reply stores/github/authorization-model.json:232]: Yep - an org owns a repo, a user owns an org

File: stores/github/authorization-model.json (hunk 4 above)
Comment (rhamzeh) [reply stores/github/authorization-model.json:19]: We don't have that in the tuples

File: stores/gdrive/authorization-model.json (hunk 1 above)
Comment (rhamzeh) [reply stores/gdrive/authorization-model.json]: Fixed

File: stores/github/authorization-model.json (hunk 2 above)
Comment (rhamzeh) [reply stores/github/authorization-model.json]: Fixed

File: stores/gdrive/authorization-model.json [hunk 5]
```diff
@@ -50,6 +63,30 @@
...
+                "type": "user"
+            }]
+          },
+          "parent": {
+            "directly_related_user_types": [{
+                "type": "folder"
+            }]
+          },
+          "viewer": {
+            "allow_public": true,
```
Comment (adriantam) [inline stores/gdrive/authorization-model.json:81]: do we still need this?

File: stores/gdrive/tuples.json [hunk 6]
```diff
@@ -20,7 +20,7 @@
     "object": "doc:public-roadmap"
```
Comment (adriantam) [inline stores/gdrive/tuples.json:20]: Should the user be 'user:*'?

File: stores/custom-roles/README.md [hunk 7]
```diff
@@ -68,39 +68,42 @@ There are users, organizations, teams and assets.
...
+    define editor: [role] or asset_editor from org
+    define org: [org]
+    define viewer: [role] or commenter or asset_viewer from org
 type asset
   relations
-    define category as self
-    define edit as self or editor from category
-    define comment as self or edit or commenter from category
-    define view as self or comment or viewer from category
+    define category: [category]
```
Comment (adriantam) [inline stores/custom-roles/README.md:103]:
```suggestion
    define category: [asset-category]
```
there is no type category

either change this to `asset-category` or change the type in line 94 to `category`

File: stores/expenses/README.md [hunk 8]
```diff
@@ -50,21 +50,26 @@ For more on modeling, check out:
 
 ### Model
 ```python
+model
+  # We are using the 1.1 schema with type restrictions
+  schema 1.1
+
```
Comment (adriantam) [inline stores/expenses/README.md:56]:
```suggestion
type user
```

**Review by adriantam — CHANGES_REQUESTED** [review]

File: stores/custom-roles/README.md (hunk 7 above)
Comment (adriantam) [reply stores/custom-roles/README.md:103]: May also need to update the JSON accordingly.

File: stores/expenses/README.md (hunk 8 above)
Comment (rhamzeh) [reply stores/expenses/README.md:56]: There is no user type in the expenses store. We have type employee instead

Comment (adriantam) [reply stores/expenses/README.md:56]: In this case, how come line 61, line 69 etc. refers to user?

Comment (rhamzeh) [reply stores/expenses/README.md:56]: Those are typos - fixed. Thanks!

**Review by adriantam — APPROVED** [review]

**Pull request merged**